src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1276,7 +1292,8 @@ SinglePartitionJoin(JoinOrderNode *currentJoinNode, TableEntry *candidateTable,
 	}
 
 	/* evaluate re-partitioning the current table only if the rule didn't apply above */
-	if (nextJoinNode == NULL && candidatePartitionMethod != DISTRIBUTE_BY_HASH)
+	if (nextJoinNode == NULL && candidatePartitionMethod != DISTRIBUTE_BY_HASH &&
```

Comment:
@marcocitus This is one of the places where I'm not sure whether the change is correct.
In general, I don't understand why we have `candidatePartitionMethod != DISTRIBUTE_BY_HASH` check here. Could you pay some extra attention to this change?